ui/lib/sync/addon/components/secrets/page/destinations.hbs

```diff
@@ -76,31 +76,31 @@
               @hasChevron={{false}}
               data-test-popup-menu-trigger
             />
-            {{#if destination.destinationPath.isLoading}}
-              <dd.Generic class="has-text-center">
-                <LoadingDropdownOption />
-              </dd.Generic>
-            {{else}}
+            <dd.Interactive
+              data-test-details
+              @route="secrets.destinations.destination.details"
+              @models={{array destination.type destination.name}}
+              @disabled={{not (has-capability @capabilities "read" pathKey="syncDestination" params=destination)}}
+            >
+              Details
+            </dd.Interactive>
+            {{#if (has-capability @capabilities "edit" pathKey="syncDestination" params=destination)}}
               <dd.Interactive
-                data-test-details
-                @route="secrets.destinations.destination.details"
+                data-test-edit
```

Comment:
Outside the scope of changes here, but I'm really trying to tighten up our test selectors. For menu items we should be using the menuItem selector in GENERAL selectors [here](https://github.com/hashicorp/vault/blob/main/ui/tests/helpers/general-selectors.ts#L37).
```suggestion
                data-test-popup-menu="edit"
```